src/include/daos_srv/control.h

```diff
@@ -14,21 +14,109 @@
 #include <stdint.h>
 #include <stdbool.h>
 #include <assert.h>
+#include <string.h>
 
-/*
+/**
  * Space separated string of CLI options to pass to DPDK when started during
  * spdk_env_init(). These options will override the DPDK defaults.
  */
 extern const char *
 dpdk_cli_override_opts;
 
-/* Device state flags */
+/** Device state flags */
 #define NVME_DEV_FL_PLUGGED	(1 << 0)
 #define NVME_DEV_FL_INUSE	(1 << 1) /* Used by DAOS (present in SMD) */
 #define NVME_DEV_FL_FAULTY	(1 << 2)
 #define NVME_DEV_FL_IDENTIFY	(1 << 3) /* SSD being identified by LED activity */
 
-/*
+/** Device state combinations */
+#define NVME_DEV_STATE_NORMAL (NVME_DEV_FL_PLUGGED | NVME_DEV_FL_INUSE)
+#define NVME_DEV_STATE_FAULTY (NVME_DEV_STATE_NORMAL | NVME_DEV_FL_FAULTY)
+#define NVME_DEV_STATE_NEW (NVME_DEV_FL_PLUGGED)
+
+static inline char *
+nvme_state2str(int dev_state)
+{
+	switch (dev_state) {
```

Comment:
It seems clunky to implement this way. Could we have the caller pass a buffer and build a string based on the bit values?

Reply:
It's a little clunky but I don't see it causing any functional problems implemented this way, it enables us to return a literal that can easily be copied into a drpc response buffer from an in-line function. Would you block on this or can I address in a follow-up PR?

Reply:
Wouldn't block on it, I see your point. If there are a limited number of valid states, it makes sense. But I feel like this maybe highlights some faults of treating the state as a singular entity to begin with. Maybe the user-level JSON output should be a series of bools representing each bit, instead of trying to translate values from a bitmap into a cohesive state represented as a string. Then it could be completely explicit.